package.json

```diff
@@ -66,7 +72,18 @@
     "build": "webpack -p --config webpack.config.prod.js",
     "watch": "webpack --config webpack.config.dev.js --watch --progress --colors",
     "test": "jest",
-    "test:watch": "jest --watch"
+    "test:watch": "jest --watch",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "linters":{
+       "*.{js}": [
+        "prettier --config .prettierrc --write",
+        "git add"
+      ],
+       "*.{ts,tsx}": ["prettier --config .prettierrc --parser typescript --write", "git add", "jest"]
```

Comment:
in `.prettierrc` we already specifying parser as typescript. So I think we don't need to specify this here